server/proxy/proxy.go

```diff
@@ -193,6 +200,100 @@ func (p *Proxy) NotifyRemoteWebhook(app *apps.App, data []byte, webhookPath stri
 	return upstream.Notify(up, creq)
 }
 
+func (p *Proxy) NotifyMessageHasBeenPosted(post *model.Post, cc *apps.Context) error {
+	postSubs, err := p.store.Subscription.Get(apps.SubjectPostCreated, cc.TeamID, cc.ChannelID)
+	if err != nil && err != utils.ErrNotFound {
+		return errors.Wrap(err, "failed to get post_created subscriptions")
+	}
+
+	subs := []*apps.Subscription{}
+	subs = append(subs, postSubs...)
+	mentions := model.PossibleAtMentions(post.Message)
+
+	botCanRead := map[string]bool{}
+	if len(mentions) > 0 {
+		appsMap := p.store.App.AsMap()
+		mentionSubs, err := p.store.Subscription.Get(apps.SubjectBotMentioned, cc.TeamID, cc.ChannelID)
+		if err != nil && err != utils.ErrNotFound {
+			return errors.Wrap(err, "failed to get bot_mentioned subscriptions")
+		}
+
+		for _, sub := range mentionSubs {
+			app := appsMap[sub.AppID]
+			if app == nil {
+				continue
```

Comment:
Should we log an error here? I was expecting subscriptions to be cleaned when an app is uninstalled and hence this case should not happen, right?

Reply:
Indeed the subscriptions are not being cleaned up when an app is uninstalled. Created a ticket for this here https://mattermost.atlassian.net/browse/MM-37930

https://github.com/mattermost/mattermost-plugin-apps/blob/dfcace445d39fa39a06bd11d8f959cacadb589f0/server/proxy/uninstall.go#L15